app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala

```diff
@@ -265,12 +312,32 @@ object SharesDisposalCYAController {
         )
         .withMarginBottom(Margin.Fixed60Bottom),
       refresh = None,
-      buttonText = parameters.mode.fold(
-        normal = "site.saveAndContinue",
-        check = "site.continue"
-      ),
-      onSubmit = routes.SharesDisposalCYAController
-        .onSubmit(parameters.srn, parameters.shareIndex, parameters.disposalIndex, parameters.mode)
+      buttonText =
+        parameters.mode.fold(normal = "site.saveAndContinue", check = "site.continue", viewOnly = "site.continue"),
```

Comment:
should viewOnly be `site.return.to.tasklist` ?

Reply:
Hi @kberec ,  QA's telling that i should go to previous page not the task list page from CYA, and in prototype also behaving the same. It should be 'site.continue'